### DISCUSSION TITLE
[help] Should the current github versions of crew, crew.cluster, and targets invalidate pipelines made with the CRAN version?

### DISCUSSION THREAD

**Discussion by kendonB** [ropensci/targets#1180] [Help] [answered]
### Help

- [X] I understand and agree to <https://books.ropensci.org/targets/help.html>.

### Description

 Should the current github versions of crew, crew.cluster, and targets invalidate pipelines made with the CRAN version?
 
 Trying to figure out which is more likely to result in a valid pipeline in future versions.

**Comment by wlandau** [comment, marked as answer]
`crew` and `crew.cluster` are independent from the persistent data storage of reproducible pipelines, so they shouldn't invalidate any targets. But there are a couple changes in the `targets` package that will unavoidably invalidate most pipelines and cause them to rerun. When that happens, I write about the details in `NEWS.md`: https://github.com/ropensci/targets/blob/main/NEWS.md#invalidating-changes. I only make these kinds of changes when I absolutely have to.

**Reply by wlandau** [reply]
It's always good to use `renv` to lock down the version of `targets` you are using for each specific project.

**Comment by kendonB** [comment]
Gotcha - I was noticing the make process hanging on Windows after moving a pipeline back from Linux (with the development versions) so reverted to CRAN. I will try to reproduce in a couple of days once deadlines pass.
